Examples/Docs/Sources/AsyncExamples/main.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+import MongoSwift
+import NIO
+
+// swiftlint:disable force_unwrapping
+
+/// Examples used for the MongoDB documentation on Causal Consistency.
+/// - SeeAlso: https://docs.mongodb.com/manual/core/read-isolation-consistency-recency/#examples
+private func causalConsistency() throws {
+    let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    let client1 = try MongoClient(using: elg)
+    defer {
+        client1.syncShutdown()
+        try? elg.syncShutdownGracefully()
+    }
+
+    // Start Causal Consistency Example 1
+    let s1 = client1.startSession(options: ClientSessionOptions(causalConsistency: true))
+    let currentDate = Date()
+    var dbOptions = DatabaseOptions(
+        readConcern: ReadConcern(.majority),
+        writeConcern: try WriteConcern(w: .majority, wtimeoutMS: 1000)
+    )
+    let items = client1.db("test", options: dbOptions).collection("items")
+    let result1 = items.updateOne(
+        filter: ["sku": "111", "end": .null],
+        update: ["$set": ["end": .datetime(currentDate)]],
+        session: s1
+    ).flatMap { _ in
+        items.insertOne(["sku": "nuts-111", "name": "Pecans", "start": .datetime(currentDate)], session: s1)
+    }
+    // End Causal Consistency Example 1
+
+    let client2 = try MongoClient(using: elg)
+
+    // Start Causal Consistency Example 2
+    let options = ClientSessionOptions(causalConsistency: true)
+    let result2: EventLoopFuture<Void> = client2.withSession(options: options) { s2 in
+        // The cluster and operation times are guaranteed to be non-nil since we already used s1 for operations above.
+        s2.advanceClusterTime(to: s1.clusterTime!)
+        s2.advanceOperationTime(to: s1.operationTime!)
+
+        dbOptions.readPreference = ReadPreference(.secondary)
+        let items2 = client2.db("test", options: dbOptions).collection("items")
+
+        return items2.find(["end": .null], session: s2).flatMap { cursor in
+            cursor.forEach { item in
+                print(item)
+            }
+        }
+    }
+    // End Causal Consistency Example 2
+}
+
+/// Examples used for the MongoDB documentation on Change Streams.
+/// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/
+private func changeStreams() throws {
+    let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    let client = try MongoClient(using: elg)
+    let db = client.db("example")
+
+    // The following examples assume that you have connected to a MongoDB replica set and have
+    // accessed a database that contains an inventory collection.
+
+    do {
+        // Start Changestream Example 1
+        let inventory = db.collection("inventory")
+        let futureCursor = inventory.watch()
```

Comment:
there's not really a way to make it compile like that, since the return types of option 1 and 2 are different - option 1 is an `EventLoopFuture<Document>` and option 2 is an `EventLoopFuture<Void>`.

if you think it's better I could duplicate the call to `watch` across the two options though to eliminate the `futureCursor` variable.

Reply:
Yeah that'd probably be better imo.

Reply:
done